Code/src/main/java/nl/utwente/group10/ui/handlers/ConnectionCreationManager.java

```diff
@@ -38,12 +51,26 @@
      */
     public static final boolean CONNECTIONS_ALLOW_TYPE_MISMATCH = true;
 
+    /** The int representing the current connection state */
+    private static int connectionState = 0; 
```

Comment:
Momenteel vermindert het invalidates in het geval van splitsende en vervolgens joinende paden van connections. Als je het weghaald kan het hierdoor dubbel gaan.
Maar, door de state te gebruiken worden loops ook nog eens extra voorkomen, mochten deze er ooit in sluipen.
En tevens is iets als dit ook mogelijk de oplossing voor het recursief aansluiten van blokken.